doc/Requirements.md

```diff
@@ -0,0 +1,374 @@
+      be represented as an array of char (string). But this would not permit to
+      check parameter validity (invariants) as well a pretty displaying the values.)
+
+Implementation MAY add add other API to access a parameter value.
+(Why: For example a C++ implementation may give access to a string as a std::string object.)
+
+##### Integers
+PF SHOULD support signed and unsigned integer parameters
+(Why: The reference implementation supports it.)
+
+PF MUST support integer with invariant size.
+(Why: It is common in C API to expect numbers to have a fixed maximum size.)
+
+The API to assess it MUST respect C abi:
+ - 2's complement for signed integer
+ - bits represent the power of two of their index.
+(Why: For easy access from C code.)
+
+Supported integer size SHOULD be at least 8, 16, 32 and 64 bits.
+(Why: The reference implementation supports it.)
+
+##### String
+PF SHOULD support array of characters.
+(Why: Everything that a computer can store fits in an array of characters.
+      It is the fallback type, if no other can be used.)
+
+The array maximum size MAY be invariant (immutable).
+(Note: This is what the reference implementation does.)
+
+The API to access the string value SHOULD support null terminated character array. As it is commonly done in C.
+(Why: For easy access from C code.)
+
+##### Fix point parameter
+PF SHOULD support fix point parameters. Ie integers divided by a fixed power of two.
+(Why: The reference implementation supports it.)
+
+The API to access the values SHOULD respect the Qm.n and UQm.n standards.
+(Why: It is the main standard for fix point parameters.)
+
+PF SHOULD support at least m + n ∈ [0,32]
+(Why: The reference implementation supports it.)
+
+##### Floating point
+PF SHOULD support floating point parameters .
+(Why: The reference implementation supports it.)
+
+The API to access the values SHOULD respect the IEEE 754 standard.
+
+PF SHOULD support at least 32 and 64 bit size floats.
+(Why: The reference implementation supports it.)
+
+##### Bit field following the C abi.
+PF SHOULD support 1 or more bit sized integers.
+(Why: The reference implementation supports it.)
+
+Such parameters called are regrouped in a so called bit parameter block.
+The API to access a bit parameter block SHOULD give access to a packed bit field.
+(Why: The reference implementation supports it.)
+
+## Identifiers
+Every parameter MUST have a identifier that uniquely identifies it.
+(Why: to identify a parameter outside the framework)
+
+This identifier SHOULD be a string.
+(Why: so that the (human) user can identify a parameter with ease.)
+
+Two PF with the same parameters MUST have the same identifier for those parameters.
+(Why: Persistence of parameter identifier across PF instances with the same configuration.
+      To identify parameters independently of the host machine and PF instance)
+
+### Parameter tree
+A parameter SHOULD be structured in a tree. Each parameter being a distinct tree leaf.
```

Comment:
I replied in the why: 
~~~
It is not a MUST because an other structure could be chose like a list or a graph.
It would not change the rest of the PF properties thank to the unique identifier requirement.
~~~